test/gtest/ucp/test_ucp_proto_mock.cc

```diff
@@ -34,71 +34,117 @@ class mock_iface {
         m_mock.cleanup();
     }
 
-    void set_mock_iface_attr(const std::string &tl_name, iface_attr_func_t cb)
+    void add_mock_iface(const std::string &dev_name, iface_attr_func_t cb)
     {
-        mock_tl(tl_name);
-        m_iface_attrs_funcs[tl_name] = cb;
+        m_iface_attrs_funcs[dev_name] = cb;
     }
 
-private:
-    void mock_tl(const std::string &tl_name)
+    void mock_transport(const std::string &tl_name)
     {
         uct_component_h component;
+
+        /* Currently only one TL can be mocked */
+        ucs_assert(nullptr == m_tl);
+
         ucs_list_for_each(component, &uct_components_list, list) {
             uct_tl_t *tl;
             ucs_list_for_each(tl, &component->tl_list, list) {
                 if (tl_name == tl->name) {
-                    EXPECT_EQ(0, m_tls.count(tl_name));
-                    m_tls[tl_name] = tl;
+                    m_mock.setup(&component->query_md_resources, query_md_mock);
+                    m_mock.setup(&tl->query_devices, query_devices_mock);
                     m_mock.setup(&tl->iface_open, iface_open_mock);
+                    m_tl = tl;
+                    return;
                 }
             }
         }
+
+        FAIL() << "Transport " << tl_name << " not found";
+    }
+
+private:
+    static ucs_status_t query_md_mock(uct_component_t *component,
+                                      uct_md_resource_desc_t **resources_p,
+                                      unsigned *num_resources_p)
+    {
+        UCS_MOCK_ORIG_FUNC(m_self->m_mock, &component->query_md_resources,
+                           component, resources_p, num_resources_p);
+        /* Keep only the first available MD */
+        *num_resources_p = ucs_min(*num_resources_p, 1);
+        return UCS_OK;
+    }
+
+    static ucs_status_t
+    query_devices_mock(uct_md_h md, uct_tl_device_resource_t **tl_devices_p,
+                       unsigned *num_tl_devices_p)
+    {
+        UCS_MOCK_ORIG_FUNC(m_self->m_mock, &m_self->m_tl->query_devices, md,
+                           tl_devices_p, num_tl_devices_p);
+        if (*num_tl_devices_p == 0) {
+            return UCS_OK;
+        }
+
+        /*
+         * Store first device name, that will be used to instantiate all the
+         * mocks. Then mock device query response to return mocks instead.
+         */
+        m_self->m_real_dev_name  = (*tl_devices_p)[0].name;
+        unsigned size            = m_self->m_iface_attrs_funcs.size();
+        auto tmp                 = (uct_tl_device_resource_t *)ucs_malloc(
+                                        sizeof(uct_tl_device_resource_t) * size,
+                                        "mock_tl_devices");
+        ucs_sys_device_t sys_dev = 1;
+        unsigned i               = 0;
+        for (const auto & it : m_self->m_iface_attrs_funcs) {
```

Comment:
```suggestion
        for (const auto &it : m_self->m_iface_attrs_funcs) {
```

Please check `format code` stage. There are relevant issues.